en/docs/install-and-setup/upgrading-wso2-api-manager/upgrading-from-300-to-310.md

```diff
@@ -0,0 +1,982 @@
+        password = "password"
+        driver = "com.ibm.db2.jcc.DB2Driver"
+        validationQuery = "SELECT 1 FROM SYSIBM.SYSDUMMY1"
+        ```
+
+4.  Update `<API-M_3.1.0_HOME>/repository/conf/deployment.toml` file as follows, to point to the correct database for user management purposes.
+
+    ```
+    [realm_manager]
+    data_source = "WSO2USER_DB"
+    ```
+
+5.  Copy the relevant JDBC driver to the `<API-M_3.1.0_HOME>/repository/components/lib` folder.
+
+    !!! info
+        In API-M 3.1.0, you do not need to configure the registry configurations as you did in the `<OLD_API-M_HOME>/repository/conf/registry.xml` file and the user database configurations as you did in in the `<OLD_API-M_HOME>/repository/conf/user-mgt.xml` file, as those configurations have been handled internally.
+
+6.  Move all your Synapse configurations to API-M 3.1.0 pack.
+    -   Move your Synapse super tenant configurations.
+        Copy the contents in the `<OLD_API-M_HOME>/repository/deployment/server/synapse-configs/default` directory and replace the contents in the `<API-M_3.1.0_HOME>/repository/deployment/server/synapse-configs/default` directory with the copied contents.
+    -   Move all your tenant Synapse configurations.
+        Copy the contents in the `<OLD_API-M_HOME>/repository/tenants` directory and replace the contents in the `<API-M_3.1.0_HOME>/repository/tenants` directory with the copied contents.
+
+    !!! warning
+        When moving the Synapse configurations, **do not replace** the following set of files as they contain some modificatiosn in API-M 3.1.0 version.
+
+        -   /api/\_RevokeAPI_.xml
+        -   /sequences/\_cors_request_handler_.xml
+        -   /sequences/main.xml
+
+    !!! attention 
+        If you are working with a **clustered/distributed API Manager setup**, follow this step on the **Gateway** node.
+
+7.  Move all your Execution plans from `<API-M_3.0.0_HOME>/repository/deployment/server/executionplans` directory to `<API-M_3.1.0_HOME>/repository/deployment/server/executionplans` directory.
+
+    !!! note
+        If you are working with a **clustered/distributed API Manager setup**, follow this step on the **Traffic Manager** node.
+
+8.  If you manually added any custom OSGI bundles to the `<API-M_3.0.0_HOME>/repository/components/dropins` directory, copy those to the `<API-M_3.1.0_HOME>/repository/components/dropins` directory. 
+
+9.  If you manually added any JAR files to the `<API-M_3.0.0_HOME>/repository/components/lib` directory, copy those and paste them in the `<API-M_3.1.0_HOME>/repository/components/lib` directory.
+
+10. WSO2 API Manager 3.1.0 has been upgraded to log4j2 (from log4j). You will notice that there is a log4j2.properties file in the `<API-M_3.1.0_HOME>/repository/conf/` directory instead of the log4j.properties file. Follow [Upgrading to Log4j2](../upgrading-wso2-api-manager/upgrading-to-log4j2.md) to migrate your existing log4j.properties file to log4j2.properties file.
```

Comment:
For the migration from 3.0.0 to 3.1.0, this step is not required I guess. Shall we remove it as Log4J2 is already in 3.0.0?

Reply:
Fixed in 38ea33a145c878e71ad0aa9069d0b137f4212c09